docs/sections/user_guide/rocoto_yaml.rst

```diff
@@ -0,0 +1,606 @@
+.. _defining_a_workflow:
+
+Defining a Rocoto Workflow
+==========================
+
+:rocoto:`Rocoto<>` is a workflow manager widely used by :ufs:`UFS<>` users and developers. It uses a custom XML language to define a set of tasks, their computational resource needs on a batch system, and their dependencies. 
+
+To date, it has been challenging to manage XML files that must support a multitude of workflow options. The ``uw rocoto`` tool defines a YAML language that can be easily manipulated like any other key/value configuration file and translates it into the syntax required by Rocoto. This enables flexibility to arbitrarily make changes to any workflow while UFS Apps can track the highest priority configurations for Community and Operational use.
+
+
+Workflow Section
+~~~~~~~~~~~~~~~~
+Starting at the top level of the YAML config for Rocoto, there are several required sections. See the example and explanation below:
+
+.. code::
+
+  workflow:
+    attrs:
+      realtime: false
+      scheduler: slurm
+    cycledef:
+      - attrs:
+          activation_offset: -06:00
+          group: howdy
+        spec: 202209290000 202209300000 06:00:00
+    entities:
+      ACCOUNT: myaccount
+      FOO: test.log
+    log: /some/path/to/&FOO;
+    tasks:
+      ...
+
+YAML Entries
+............
+
+``attrs``: This section allows users to define any number of the available attributes available to the workflow tag in Rocoto's native XML language. Set them with a key/value that matches the exact syntax needed by Rocoto. These attributes work to fill in the ``<workflow>`` tag. For example,
```

Comment:
- "available attributes available" -- pick one "available" :)
- "workflow tag"` -> "`<workflow>` element" ? I think using XML syntax helps, and IMO we should be consistent about using XML terminology (i.e. "element") for XML concepts.
- "Rocoto's native XML language" -> "Rocoto XML" for brevity?
- "Set them" -> "Set each" for singular/plural agreement with what follows in the sentence.

Reply:
Hmm, I think maybe I'm being too limited about "tag" vs "element". It looks like "tag" is correct w.r.t. say `<tag>` (an _opening_ tag), or `</tag>` (a _closing_ tag), or `<tag key="value"/>` when there aren't separate opening/closing tags. I think "element" would describe the whole "thing" identified by a tag: Maybe the whole block between e.g. `<workflow> ... </workflow>`, or a one-liner like `<tag key="value" />`.

So, "`<workflow>` tag" seems correct if referring to that bit of syntax, if that's what meant. It wouldn't hurt to use precise terminology.

cf. https://stackoverflow.com/questions/12027770/what-is-the-difference-between-a-tag-and-an-element

Reply:
Do you think it would be worth defining that at the top? I don't think everybody reading this would know about the specific XML terminology here.

Reply:
It wouldn't hurt but, on the other hand, as long as we use the terms consistently and correctly, I think most readers will quickly come to understand what they mean, and they can easily google "XML terminology" if they're still confused. I don't think we're obliged to define existing terms-of-art, or write a woefully incomplete XML primer. If we found a nice page defining XML terms, a link to _that_ at the top might be nice -- a quick click for the reader who wants to brush up.